src/components/wallets/WalletTransactionList.tsx

```diff
@@ -0,0 +1,244 @@
+import { useEffect } from 'react'
+import styled from 'styled-components'
+import { DateTime } from 'luxon'
+import { gql } from '@apollo/client'
+
+import { theme } from '~/styles'
+import { GenericPlaceholder } from '~/components/GenericPlaceholder'
+import ErrorImage from '~/public/images/maneki/error.svg'
+import EmptyImage from '~/public/images/maneki/empty.svg'
+import { Avatar, Button, Icon, Skeleton, Typography } from '~/components/designSystem'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import {
+  useGetWalletTransactionsLazyQuery,
+  Wallet,
+  WalletTransactionStatusEnum,
+  WalletTransactionTransactionTypeEnum,
+} from '~/generated/graphql'
+import { intlFormatNumber } from '~/core/intlFormatNumber'
+
+gql`
+  query getWalletTransactions($walletId: ID!, $page: Int, $limit: Int) {
+    walletTransactions(walletId: $walletId, page: $page, limit: $limit) {
+      metadata {
+        currentPage
+        totalPages
+        totalCount
+      }
+      collection {
+        id
+        status
+        amount
+        creditAmount
+        settledAt
+      }
+    }
+  }
+
+  fragment WalletInfosForTransactions on Wallet {
+    currency
+  }
+`
+
+interface WalletTransactionListProps {
+  isOpen: boolean
+  wallet: Omit<Wallet, 'updatedAt'>
+}
+
+export const WalletTransactionList = ({ isOpen, wallet }: WalletTransactionListProps) => {
+  const { translate } = useInternationalization()
+  const [getWalletTransactions, { data, error, fetchMore, loading, refetch }] =
+    useGetWalletTransactionsLazyQuery({
+      variables: { walletId: wallet.id, limit: 20 },
+    })
+  const list = data?.walletTransactions?.collection
+  const { currentPage = 0, totalPages = 0 } = data?.walletTransactions?.metadata || {}
+  const hasData = !!list && !!list?.length
+
+  useEffect(() => {
+    if (isOpen && !data && !loading && !error) {
```

Comment:
If there is an error, you could reload the query no ?

Reply:
hum yes, but error will trigger the `refecth` method and not use this effect 🤔 

So I guess it's ok to keep this `!error` condition here otherwise it will refetch automatically on error and not manually as expected WDYT ?

Reply:
ok make sense :)